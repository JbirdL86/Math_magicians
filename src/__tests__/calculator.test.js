import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Calculator events works', () => {
  test('Clicking 2 is displayed', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '2' }));
    expect(screen.getByTestId('display')).toContainHTML('2');
  });

  test('Clicking 2 + 3 is displayed', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '+' }));
    fireEvent.click(screen.getByRole('button', { name: '3' }));
    expect(screen.getByTestId('display')).toContainHTML('2 + 3');
  });

  test('Clicking 2 + 3 = display 5', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '+' }));
    fireEvent.click(screen.getByRole('button', { name: '3' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));
    expect(screen.getByTestId('display')).toContainHTML('5');
  });
});
