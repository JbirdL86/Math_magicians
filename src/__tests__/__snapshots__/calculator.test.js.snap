// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calc_main_container"
>
  <h1
    className="calc_title_text"
  >
    Let's do some math!
  </h1>
  <div
    className="calc_container"
  >
    <div
      className="result"
    >
      <p
        data-testid="display"
      >
         
         
      </p>
    </div>
    <div
      className="calc-buttons"
    >
      <div
        className="left-btns-container"
      >
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="left-btn-cero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="left-btn"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
      <div
        className="right-btns-container"
      >
        <button
          className="right-btn"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="right-btn"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="right-btn"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="right-btn"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="right-btn"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
